### PR TITLE
[backport] Fix fp16 issue in batch normalization + Keep backward compatibility on cupy.cudnn.batch_normalization_forward_training

### PR DESCRIPTION
**Pull request by toslunar**
Backport of #2060 and #2072.

### CODE REVIEW

**Comment by hvy** [comment]
@takagi Can you take a look at this backport?

**Comment by takagi** [comment]
LGTM!
